app/assets/javascripts/discourse/app/components/footer-nav.gjs

```diff
@@ -0,0 +1,183 @@
+import { service } from "@ember/service";
+import { modifier as modifierFn } from "ember-modifier";
+import DButton from "discourse/components/d-button";
+import concatClass from "discourse/helpers/concat-class";
+import htmlClass from "discourse/helpers/html-class";
+import { postRNWebviewMessage } from "discourse/lib/utilities";
+import { SCROLLED_UP, UNSCROLLED } from "discourse/services/scroll-direction";
+import not from "truth-helpers/helpers/not";
+
+class FooterNav extends Component {
+  @service appEvents;
+  @service capabilities;
+  @service scrollDirection;
+  @service composer;
+  @service modal;
+
+  @tracked shouldToggleMobileFooter = false;
+  @tracked canGoBack = false;
+  @tracked canGoForward = false;
+
+  currentRouteIndex = 0;
+  routeHistory = [];
+  backForwardClicked = false;
+
+  registerAppEvents = modifierFn(() => {
+    this.appEvents.on("page:changed", this, "_routeChanged");
+
+    return () => {
+      this.appEvents.off("page:changed", this, "_routeChanged");
+    };
+  });
+
+  @action
+  _routeChanged(route) {
+    // only update route history if not using back/forward nav
+    if (this.backForwardClicked) {
+      this.backForwardClicked = null;
+      return;
+    }
+
+    this.routeHistory.push(route.url);
+    this.currentRouteIndex = this.routeHistory.length;
+    this.setBackForward();
+  }
+
+  _modalOn() {
+    const backdrop = document.querySelector(".modal-backdrop");
+    if (!backdrop) {
+      return;
+    }
+
+    postRNWebviewMessage(
+      "headerBg",
+      getComputedStyle(backdrop)["background-color"]
+    );
+  }
+
+  _modalOff() {
+    const dheader = document.querySelector(".d-header");
+    if (!dheader) {
+      return;
+    }
+
+    postRNWebviewMessage(
+      "headerBg",
+      getComputedStyle(dheader)["background-color"]
+    );
+  }
+
+  @action
+  setDiscourseHubHeaderBg(hasAnActiveModal) {
+    if (!this.capabilities.isAppWebview) {
+      return;
+    }
+
+    if (hasAnActiveModal) {
+      this._modalOn();
+    } else {
+      this._modalOff();
+    }
+  }
+
+  @action
+  dismiss() {
+    postRNWebviewMessage("dismiss", true);
+  }
+
+  @action
+  share() {
+    postRNWebviewMessage("shareUrl", window.location.href);
+  }
+
+  @action
+  goBack() {
+    this.currentRouteIndex = this.currentRouteIndex - 1;
+    this.setBackForward();
+    this.backForwardClicked = true;
+    window.history.back();
+  }
+
+  @action
+  goForward() {
+    this.currentRouteIndex = this.currentRouteIndex + 1;
+    this.setBackForward();
+    this.backForwardClicked = true;
+    window.history.forward();
+  }
+
+  get isVisible() {
+    return (
+      [UNSCROLLED, SCROLLED_UP].includes(
+        this.scrollDirection.lastScrollDirection
+      ) && !this.composer.isOpen
+    );
+  }
+
+  setBackForward() {
+    this.canGoBack =
+      this.currentRouteIndex > 1 || document.referrer ? true : false;
+    this.canGoForward =
+      this.currentRouteIndex < this.routeHistory.length ? true : false;
+  }
```

Comment:
Lots of state being juggled here. I think `canGoBack` and `canGoForward` should be declarative getters (and to make that work, routeHistory should be a TrackedArray)

Reply:
although... at some point, currentRoutedIndex is set arbitrirarly

Reply:
Yes ok I can make that work